utils/src/deployer.rs

```diff
@@ -332,9 +333,11 @@ pub async fn upgrade_light_client_v2(
             } else {
                 LightClientV2::BYTECODE.encode_hex()
             };
-            let lcv2_linked_bytecode = Bytes::from_hex(
-                target_lcv2_bytecode.replace(LIBRARY_PLACEHOLDER_ADDRESS, &pv2_addr.encode_hex()),
-            )?;
+            let lcv2_linked_bytecode = Bytes::from_hex(target_lcv2_bytecode.replacen(
+                LIBRARY_PLACEHOLDER_ADDRESS,
+                &pv2_addr.encode_hex(),
+                1,
+            ))?;
```

Comment:
I guess to actually make this error we need to check if the string is still there after the first replacement.

Reply:
you are right, i'll fix that

Reply:
done in [dae9d5c](https://github.com/EspressoSystems/espresso-network/pull/2706/commits/dae9d5cf3d9ec241696693e70f2e10e8fd0f02b9)